### PR TITLE
New homepage

### PR DESCRIPTION
**Pull request by sethladd**
Fixes https://github.com/flutter/flutter/issues/9671
Fixes https://github.com/flutter/flutter/issues/9609
Fixes https://github.com/flutter/flutter/issues/9608
Fixes https://github.com/flutter/flutter/issues/9578
Fixes https://github.com/flutter/flutter/issues/8896
Fixes https://github.com/flutter/flutter/issues/8718
Fixes https://github.com/flutter/flutter/issues/9300
Fixes https://github.com/flutter/flutter/issues/9623

### CODE REVIEW

**Comment by Hixie** [comment]
feedback from someone looking at this on mobile: there's no left margin, the text starts at pixel 0

**Comment by sethladd** [comment]
Thanks!

**Comment by Hixie** [comment]
If I make my page narrow on desktop, the text goes off the left edge entirely.

**Comment by jhirth** [comment]
There is no prominent link to the FAQ (/faq/). Was this a deliberate omission?

**Comment by sethladd** [comment]
Good catch on the FAQ, thanks!

**Comment by sethladd** [comment]
Added FAQ.

Known issue: the footer behaves weirdly sometimes. Will address in future PR.

Identified the source of the non-responsiveness. Can fix in this PR.

Can fix the top toolbar not responding soon enough in this PR.

File: index.md [hunk 1]
```diff
@@ -1,213 +1,270 @@
...
+                rich motion, smooth scrolling, and
+                beautiful customizable components.
+                </p>
+            </div>
+
+            <div class="homepage__key_point">
+                <div class="homepage__key_point_title">Used by Google</div>
+
+                <p>
+                Flutter is used by Google, works with Firebase
```
Comment (InMatrix) [inline index.md]: Maybe it's worth saying Flutter is used by Google and others? I'm a little worried that some people wonder if it's used by Google _only_.

File: index.md [hunk 2]
```diff
@@ -1,213 +1,270 @@
...
+
+                <p>
+                Delight your users and make your brand stand out with
+                rich motion, smooth scrolling, and
+                beautiful customizable components.
+                </p>
+            </div>
+
+            <div class="homepage__key_point">
+                <div class="homepage__key_point_title">Used by Google</div>
```
Comment (InMatrix) [inline index.md]: Will "Used in production apps" better capture the content (e.g., Firebase and other SDKs) below this heading?

File: index.md [hunk 3]
```diff
@@ -1,213 +1,270 @@
...
+                and other mobile app SDKs,
+                and is open source.
+                </p>
+            </div>
+
+        </div>
+    </section>
+
+    <section class="homepage__hot_reload">
+        <h1>Fast development</h1>
```
Comment (InMatrix) [inline index.md]: Adding the lightening bolt icon next to "Fast development" is likely to help users discover the feature, once they are in the IDE.

Comment (sethladd) [inline index.md]: Can you link me to the icon that we have the rights to use? Happy to add it here. Bonus if it's an SVG that I can embed :)

File: index.md [hunk 4]
```diff
@@ -1,213 +1,270 @@
...
+        </div>
+    </section>
+
+    <section class="homepage__beautiful_uis">
+        <h1>Expressive, beautiful UIs</h1>
+
+        <p>
+        Delight your users with Flutter's built-in
+        beautiful Material Design and
+        Cupertino (iOS-flavor) widgets, rich motion APIs,
```
Comment (InMatrix) [inline index.md]: Can we have GIF or embedded video to show "motion" and smooth scrolling? At the minimum, a link to a video demo on YouTube would help support our claims.

Comment (sethladd) [inline index.md]: Totally. I want to try to use HTML5 video for this.

File: index.md [hunk 5]
```diff
@@ -1,213 +1,270 @@
...
+        Easily compose your UI with Flutter's
+        modern functional-reactive framework and
+        rich set of platform, layout, and foundation widgets.
+        Solve your tough UI challenges with
+        powerful and flexible APIs for 2D, animation, gestures,
+        effects, and more.
+        </p>
+
+{% prettify dart %}
+class MyMessage extends StatelessWidget {
```
Comment (InMatrix) [inline index.md]: This snippet is short and sweet, but it doesn't show the reactiveness of the framework. How about use this snippet: https://flutter.io/widgets-intro/#changing-widgets-in-response-to-input?

File: index.md [hunk 6]
```diff
@@ -1,213 +1,270 @@
...
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Text('Hello, world!'),
+    );
+  }
+}
+{% endprettify %}
+
+        <p>
+        Browse the <a href="/widgets/">widget catalog</a>
```
Comment (InMatrix) [inline index.md]: It might be more useful to have the link to the Widget Catalog in the "Expressive, beautiful UIs" section.

File: index.md [hunk 7]
```diff
@@ -1,213 +1,270 @@
...
-Flutter is a mobile app SDK for building high-performance, high-fidelity,
-apps for iOS and Android, from a single codebase.
+        </div>
 
-The goal is to enable developers to deliver high-performance apps that feel natural
-on different platforms. We embrace differences in scrolling behaviors,
-typography, icons, and more.
+        <p>
+        Learn more about what makes Flutter special in the
+        <a href="/technical-overview/">technical overview</a>.
```
Comment (InMatrix) [inline index.md]: This page currently doesn't show up in the side navigation.

File: index.md [hunk 8]
```diff
@@ -1,213 +1,270 @@
...
-No mobile development experience is required to get started. Apps are written
-in [Dart](https://dartlang.org/), which will look familiar if you've used a
-language like Java or JavaScript. Experience with object-oriented languages
-is definitely helpful, but even non-programmers have made Flutter apps!
+        <div class="homepage__try_today">Try Flutter today. Getting started is easy.</div>
 
-## Why use Flutter?
+        <div class="homepage__button_row">
+        <a href="/getting-started/" class="get-started-button">Get Started</a>
+        {% include note.html content="_Flutter is an alpha, open-source project_." %}
```
Comment (InMatrix) [inline index.md]: Add something about how to give up feedback here?

**Comment by InMatrix** [comment]
@sethladd Thank you for putting this new homepage together. It looks very nice overall. Please see my inline comments for some potential tweaks.

**Comment by christian-lefty** [comment]
I would suggest that most of the stuff in the Unified app development section be links to relevant doc

**Comment by sethladd** [comment]
> I would suggest that most of the stuff in the Unified app development section be links to relevant doc

Thanks! One of the things that this exercise is pointing out is that we need more docs for some things, and to bring some docs over to flutter. :)

File: index.md (hunk 3 above)
Comment (InMatrix) [inline index.md]: @devoncarew Do you have the source image of the Hot Reload icon?

Comment (devoncarew) [inline index.md]: we have a 32x32 png (https://github.com/flutter/flutter-intellij/blob/master/resources/icons/hot-reload%402x.png)

If you want a larger image, there is one that's similar but not identical which I could dig up.

File: index.md (hunk 8 above)
Comment (sethladd) [inline index.md]: We used to have that, but now there's a Support link on every page on the website.

File: index.md (hunk 7 above)
Comment (sethladd) [inline index.md]: Good catch. Added.

File: index.md (hunk 6 above)
Comment (sethladd) [inline index.md]: Can do.

File: index.md (hunk 3 above)
Comment (sethladd) [inline index.md]: thanks!

File: index.md (hunk 1 above)
Comment (sethladd) [inline index.md]: Added "and others"

File: index.md (hunk 2 above)
Comment (sethladd) [inline index.md]: added "in production"

File: index.md [hunk 9]
```diff
@@ -1,213 +1,297 @@
...
+
+        <p>
+        Make your app come to life
+        with platform APIs, 3rd party SDKs,
+        and native code.
+        Reuse your existing Java, Swift, and ObjC code.
+        </p>
+
+{% prettify dart %}
+// From https://github.com/flutter/flutter/tree/master/examples/platform_channel
```
Comment (mit-mit) [inline index.md]: Maybe move the 'From' link outside the code section, and make it a link to the sample.

File: index.md [hunk 10]
```diff
@@ -1,213 +1,297 @@
...
+        Flutter has the tools and libraries to help you easily
+        bring your ideas to life on iOS and Android.
+        If you don't have any mobile development experience, Flutter
+        is an easy and fast way to build beautiful mobile apps.
+        If you are an experienced iOS or Android developer,
+        you can use Flutter for your views and leverage much of your
+        existing Java/ObjC/Swift investment.
+        </p>
+
+        <div class="feature-lists">
```
Comment (mit-mit) [inline index.md]: These lists look a little odd. Maybe they should be links?

File: index.md [hunk 11]
```diff
@@ -1,213 +1,297 @@
 ---
-title: Welcome to Flutter
-sidebar: home_sidebar
+title: Easy and fast SDK for beautiful apps
```
Comment (sethladd) [inline index.md]: maybe? do people think apps != mobile, these days?

File: support.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
+---
+title: Support
+layout: page
+permalink: /support/
+---
+
+We're open source and would love to hear from you!
+
+We encourage you to reach out and
```
Comment (sethladd) [inline support.md]: short lines are fantastic for tech writer reviews. I was trained to do this :)

File: index.md [hunk 13]
```diff
@@ -1,213 +1,297 @@
...
+    });
+  }
+
+  Widget build(BuildContext context) {
+    // This method is rerun every time setState is called.
+    // The Flutter framework has been optimized to make rerunning
+    // build methods fast, so that you can just rebuild anything that
+    // needs updating rather than having to individually change
+    // instances of widgets.
+    return new Row(
```
Comment (sethladd) [inline index.md]: Good catch. I think I took this from another page on our site.

File: index.md (hunk 9 above)
Comment (sethladd) [inline index.md]: I think I tried that, but I'm happy to try again.

File: index.md (hunk 10 above)
Comment (sethladd) [inline index.md]: One insight from this exercise is that we have a ton of awesome capabilities in our system, but we don't always have docs that link to the specific feature or capability. Agreed that many of these items can and should be links. I'll try to find links for the subjects that do have a natural documentation for them.

**Comment by sethladd** [comment]
@mit-mit not sure I'm going to keep the "cards". If we do, agreed we should make them consistent.

**Comment by sethladd** [comment]
@InMatrix @eseidelGoogle @mit-mit I'm at the diminishing returns stage of this PR. Can you PTAL ?

After this lands, I want to fix the code samples on small screens. They should scroll horizontally. And, add more links for the feature list.

**Comment by eseidelGoogle** [comment]
Do I need to wait for the stage bot to do it's thing?

**Comment by sethladd** [comment]
It's all set: https://flutter-io-deploy-two.firebaseapp.com/

**Comment by eseidelGoogle** [comment]
The code example in the "Modern, reactive framework" doesn't feel very compelling.
The "Unified app development" section has a wall of buzzwords, but I'm not 100% sure what it's telling me (SEO?)

But honestly this is *infinitely* better than what we have today, so I would just press the button and ship it.  lgtm.  We will iterate from here.

**Comment by sethladd** [comment]
> doesn't feel very compelling

Any thoughts on would would make it more compelling? We can write something custom for the homepage.

cc @InMatrix 

> wall of buzzwords

This is intended to do a few things:

1. Convince new users that we're "full stack". We have what you need.
1. SEO
1. Provide a basis for further docs and links

File: index.md (hunk 13 above)
Comment (mit-mit) [inline index.md]: Looks like this still isn't fixed?

**Comment by mit-mit** [comment]
My only remaining concern is that the code snippet shown under Modern, reactive framework still isn't valid code, but I'd be to see a bug opened for that and then handle that post merge.

LGTM

File: index.md (hunk 13 above)
Comment (sethladd) [inline index.md]: Not clear this needs to have a Material. Can you have a Material inside another Material? This widget is intended to be inside another build() of something that has material.

But, we'll merge, and chat about a better sample.

**Comment by sethladd** [comment]
Thanks for taking a look!

**Comment by InMatrix** [comment]
Sorry about the delayed response. Yes, LGTM. As I mentioned previously, I'd love to see animations to show our performance and motion capability in the "Expressive, beautiful UIs" card.

**Pull request merged**